xml/upgrade-3.10-1.xsl

```diff
@@ -126,7 +126,13 @@
 <!-- Name/value pairs -->
 
 <!-- Drop any nvpair that does not have a value attribute -->
-<xsl:template match="nvpair[not(@value)]"/>
+<xsl:template match="nvpair[not(@value)]">
+    <xsl:call-template name="warning">
+        <xsl:with-param name="msg"
+                        select="concat('Dropping nvpair ', @id,
+                                       ' without a value')"/>
```

Comment:
Could this cause a behavioral change? If we would just have ignored it anyway, we can dispense with the message, but if there's some circumstance where it could have reset to the default previously, then the warning is good

Reply:
It doesn't look like it... if an nvpair has no name or no value, then the unpack function skips it (even in 2.1.8). The transformation pipeline drops valueless nvpairs before removing duplicates.